### PR TITLE
Moments page for Root Certification expiration (1 of 6)

### PR DESCRIPTION
**Pull request by reemhamz**
## One-line summary
Moments page made for letting users know to update their Firefox browsers so they aren't affected by the Root Certification expiration.

This is 1 of 6 moments pages to be developed for the Root Certification info. More to come in a few months

📆  **Launch date: October 9th 2024** 📆 

>[!NOTE]
> This is hard-coded in 15 languages, enjoy the review! 😅 

## Issue / Bugzilla link
https://github.com/mozilla/bedrock/issues/15162

### Localhost

- [ ] http://localhost:8000/en-US/firefox/welcome/19/
- [ ] http://localhost:8000/de/firefox/welcome/19/
- [ ] http://localhost:8000/fr/firefox/welcome/19/
- [ ] http://localhost:8000/es-ES/firefox/welcome/19/
- [ ] http://localhost:8000/et/firefox/welcome/19/
- [ ] http://localhost:8000/el/firefox/welcome/19/
- [ ] http://localhost:8000/pl/firefox/welcome/19/
- [ ] http://localhost:8000/cs/firefox/welcome/19/
- [ ] http://localhost:8000/zh-CN/firefox/welcome/19/
- [ ] http://localhost:8000/pt-BR/firefox/welcome/19/
- [ ] http://localhost:8000/ar/firefox/welcome/19/
- [ ] http://localhost:8000/ru/firefox/welcome/19/
- [ ] http://localhost:8000/ja/firefox/welcome/19/
- [ ] http://localhost:8000/id/firefox/welcome/19/
- [ ] http://localhost:8000/hu/firefox/welcome/19/

### Demo servers
- [ ] https://www-demo6.allizom.org/en-US/firefox/welcome/19/
- [ ] https://www-demo6.allizom.org/de/firefox/welcome/19/
- [ ] https://www-demo6.allizom.org/fr/firefox/welcome/19/
- [ ] https://www-demo6.allizom.org/es-ES/firefox/welcome/19/
- [ ] https://www-demo6.allizom.org/et/firefox/welcome/19/
- [ ] https://www-demo6.allizom.org/el/firefox/welcome/19/
- [ ] https://www-demo6.allizom.org/pl/firefox/welcome/19/
- [ ] https://www-demo6.allizom.org/cs/firefox/welcome/19/
- [ ] https://www-demo6.allizom.org/zh-CN/firefox/welcome/19/
- [ ] https://www-demo6.allizom.org/pt-BR/firefox/welcome/19/
- [ ] https://www-demo6.allizom.org/ar/firefox/welcome/19/
- [ ] https://www-demo6.allizom.org/ru/firefox/welcome/19/
- [ ] https://www-demo6.allizom.org/ja/firefox/welcome/19/
- [ ] https://www-demo6.allizom.org/id/firefox/welcome/19/
- [ ] https://www-demo6.allizom.org/hu/firefox/welcome/19/

### CODE REVIEW

**Review by janbrasna — CHANGES_REQUESTED** [review]
👋 Enjoyed! ;)

_(BTW, what is "addon-geddan" in the commit message?)_

File: bedrock/firefox/templates/firefox/welcome/page19.html
Comment (reemhamz) [reply bedrock/firefox/templates/firefox/welcome/page19.html:92]: The team confirmed this should be uppercase

File: bedrock/firefox/templates/firefox/welcome/page19.html
Comment (reemhamz) [reply bedrock/firefox/templates/firefox/welcome/page19.html:204]: Also confirmed to be uppercase

File: bedrock/firefox/templates/firefox/welcome/base.html
Comment (reemhamz) [reply bedrock/firefox/templates/firefox/welcome/base.html:29]: nice one! my brain was pooped while working on this bit

File: bedrock/firefox/templates/firefox/welcome/page18.html
Comment (reemhamz) [reply bedrock/firefox/templates/firefox/welcome/page18.html:82]: ah, nope! I cut this by accident.

File: bedrock/firefox/urls.py
Comment (janbrasna) [inline bedrock/firefox/urls.py:173]: This should now pass ruff, sry, the suggestion ui ate my whitespace so i closed it wrong:
```suggestion
    ),
```

File: bedrock/firefox/templates/firefox/welcome/base.html
Comment (janbrasna) [inline bedrock/firefox/templates/firefox/welcome/base.html:23]: Or even better, there's one already in img/logos:

```suggestion
        <img src="{{ static('img/logos/firefox/logo-word-hor.svg') }}" alt="Firefox" width="120" height="40" class="c-header-logo">
```

(that also has the correct dimensions iirc)

File: media/img/firefox/welcome/firefox-wordmark-hor.svg
Comment (janbrasna) [inline media/img/firefox/welcome/firefox-wordmark-hor.svg:1]: attn: welcome/16 and /18 both reference the file, so better not remove that from those…

_(i have a wip in my backlog to clean these welcome xrefs between versions where possible, and also dedupe the assets if they're already somewhere more canonical… so i'll tackle that at some point later separately.)_

File: bedrock/firefox/templates/firefox/welcome/base.html
Comment (janbrasna) [inline bedrock/firefox/templates/firefox/welcome/base.html:25]:
```suggestion
      {% block warning_notification %}{% endblock %}
```

**Review by janbrasna — CHANGES_REQUESTED** [review]
Following up with a quick linting fixup, and a broken logotype on page16-page18 when switching dark/light modes:

File: bedrock/firefox/templates/firefox/welcome/page19.html
Comment (janbrasna) [reply bedrock/firefox/templates/firefox/welcome/page19.html]: Yea reckon the copy has a stamp of approval — just offered the sentence case when I happen to have the actual accents on my keyboard;D just in case…

File: media/img/firefox/welcome/firefox-wordmark-hor.svg
Comment (reemhamz) [reply media/img/firefox/welcome/firefox-wordmark-hor.svg:1]: Those pages are mentioning `page../firefox-wordmark-light.svg`, not this particular file path?

File: bedrock/firefox/templates/firefox/welcome/base.html
Comment (reemhamz) [reply bedrock/firefox/templates/firefox/welcome/base.html:23]: Ah wow thanks for finding this! I'll implement this logo for pages 16-18 dark/light mode as well 🌞🌚

File: media/img/firefox/welcome/firefox-wordmark-hor.svg
Comment (janbrasna) [reply media/img/firefox/welcome/firefox-wordmark-hor.svg:1]: At one point this image was _moved_ here from page16 assets so I figured it would be missing there:

Screen Shot 2024-09-26 at 05 34 43

Now that 16–18 are updated too to link the canonical location instead it should be all hunky-dory. (Just wanted to point out it touches more welcome pages' both light and dark appearances so to make sure they're also included in testing they still render as expected…)

Comment (reemhamz) [reply media/img/firefox/welcome/firefox-wordmark-hor.svg:1]: d'oh, i forgot that i moved it, i thought I just pasted it! I was looking for that file during this fix 😅 thanks for pointing it out to me. I also just went ahead and used the global image you pointed out in a previous comment to replace all these individual page-specific assets, so it'll be more uniform going forward :)

Comment (janbrasna) [reply media/img/firefox/welcome/firefox-wordmark-hor.svg:1]: 🙏 Praise! Thanks, lovely. 💟 Those assets linked between page versions that have nothing common are always in risk being inadvertently removed breaking something unrelated _(here, here, me, me…)_ so this was def on my backlog to make more universal, thanks for taking care of it going few versions back (and getting the correct dimensions! I know the local assets were not the exact right size all the time…)

**Review by craigcook — APPROVED** [review]
This all looks good to me, nice work! r+ but it sounds like they want to review on a demo server before we merge? Though I'm not sure what needs review, we can't really make copy changes since it's all translated already... but we can merge as soon as we get signoff.

**Comment by reemhamz** [comment]
Can do -- I'll set up demos now :) !

**Review by alexgibson — COMMENTED** [review]
Drive by note:

If they want to measure the number of actual installs this page generates (which I'm assuming they do), then we probably want to load the stub attribution script on welcome pages, by removing this override:

https://github.com/mozilla/bedrock/blob/main/bedrock/firefox/templates/firefox/welcome/base.html#L59-L60

**Comment by alexgibson** [comment]
Sound like you can ignore my comment above, they’re just gonna rely on Firefox version update metrics instead 👍

**Pull request merged**
